types/src/error.rs

```diff
@@ -63,11 +62,16 @@ impl<'a> ErrorResponse<'a> {
 		ErrorResponse { jsonrpc: self.jsonrpc, error: self.error.into_owned(), id: self.id.into_owned() }
 	}
 
-	/// Get the [`ErrorObject`] of the error response.
-	pub fn error_object(&self) -> &ErrorObject {
+	/// Get a reference to the [`ErrorObject`] of the error response.
+	pub fn as_error_object(&self) -> &ErrorObject {
 		&self.error
 	}
 
+	/// Consume the error response and extract the [`ErrorObject`].
+	pub fn into_error_object(self) -> ErrorObject<'a> {
+		self.error
+	}
+
```

Comment:
I wonder whether it's better to return `ErrorObject<'_>` or `&'_ ErrorObject<'a>` here; one could do this for instance:

```rust
/// Get the [`ErrorObject`] of the error response with a lifetime tied to self.
pub fn as_error_object(&self) -> ErrorObject<'_> {
	ErrorObject {
		code: self.error.code,
		message: StdCow::Borrowed(self.error.message.borrow()),
		data: self.error.data.as_ref().map(|d| StdCow::Borrowed(d.borrow())),
	}
}
```

But yeah I don't know! Looks good as is; just pondering :)

Reply:
good idea, I like it

Reply:
We could just clone the `Cow` instead to get `'a` but maybe a bit weird in `fn as_error_object`

I think this should be what most users want to anyway,

Reply:
Cloning has the downside of _maybe_ allocating and maybe not depending on the original error state (if we knew the error was always borrowed I'd probably go for that). With your API it's explicit at least, so yeah I think it's all good the way you did it!